### PR TITLE
Add support for existing aml environment for aml system

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

Add support for existing aml environment for aml system. Users can specify env name, version and label to for their Environment.

## Checklist before requesting a review
- [x] Add unit tests for this change.
- [x] Make sure all tests can pass.
- [x] Update documents if necessary.
- [x] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/systems/common.py [hunk 1]
```diff
@@ -37,6 +37,12 @@ def _get_abspath(cls, v):
             return None
 
 
+class AzureMLEnvironmentConfig(ConfigBase):
```
Comment (guotuofeng) [inline olive/systems/common.py:40]: is it better to merge this class with AzureMLDockerConfig by renaming it to AzureMLEnvironmentConfig?

Comment (guotuofeng) [reply olive/systems/common.py:40]: Basically, the two classes' behavior stand for get/create environment used by aml.

Comment (guotuofeng) [reply olive/systems/common.py:40]: @jambayk, what's your opinion?

Comment (xiaoyu-work) [reply olive/systems/common.py:40]: That makes sense however it will be confusing between get and create. `aml_docker_config` currently will create an anonymous environment, when creating a new environment, you can also pass same arguments as `get` like name/version/label. This reminds me that we could improve `create` feature that user can reuse the previous creating environment by also accepting name/version/label for `create` operation.

Comment (jambayk) [reply olive/systems/common.py:40]: I think since this has two different types of environments:
1. annonymous environment created with dockerconfig
2. pre-existing environment that we reference using name,version,label

I think it is better to have separate config for each. Otherwise combining them makes it harder for the user to understand the difference between the two use cases since all the config parameters will be together even though they don't have any common parameters.

Comment (xiaoyu-work) [reply olive/systems/common.py:40]: We should improve our logic in either way:

- If we want to keep the new creating environment anonymous, we may need to make sure we delete it after this run (we don't do this now). And additionally, supporting using existing Environment by specifying its name/version like what this PR is doing.
- If not, that would be a new feature in Olive: Olive supports creating a long-lasting Azure ML Environment without deleting it after Olive run, with this, user can also provide name/version to this new environment and Olive creates a new env with name/version so they can use it in the future.

@devang-ml what do you think?

Comment (jambayk) [reply olive/systems/common.py:40]: I don't think there is a need to delete the anonymous environment or name the environment. From what I understand about aml environments, the environments are hashed by the contents of the environment (dockerfile, etc) and backed by an acr docker image. 
This allows multiple aml runs to use the same cached environment (docker image in their backend) as long as they have the same environment config. You can see of the environments used by our ci tests is using an anonymous environment that was build months ago. 
image

I don't think deleting an "environment" actually deletes any of the underlying data. Probably just the reference to it (name, label, version). If it actually deletes it, then we would have to build the environment from scratch everytime which is time consuming. The caching behavior of aml is one of its advantages.

tldr: I think naming an environment is just creating a reference to underlying hashed + cached data (build environment, environment docker image). so there is no need for us to worry about naming and managing environment ourselves. 
I think it is enough to allow user to specify their own named environments if they want.

Comment (guotuofeng) [reply olive/systems/common.py:40]: I agree with @jambayk that we don't need to delete the anonymous envs. This comments can be treated as open discussion since I am not sure which approach is better. Maybe we can wait for comments from @devang-ml

Comment (xiaoyu-work) [reply olive/systems/common.py:40]: There was one issue happened to me before: if specifying Olive repo in dockerfile like "git+https://github.com/microsoft/Olive#egg=olive-ai", even there will be new version and building the environment again, it will always use old main branch commit to build the environment (I guess somehow AML cache system used old build cache), and did not fetch the latest commit for Olive main branch. I don't see any user reporting this but this could be a potential issue.

Comment (guotuofeng) [reply olive/systems/common.py:40]: This is because AML build environment using the conda.yaml. and it doesn't check whether the external link changes.

Comment (devang-ml) [reply olive/systems/common.py:40]: Conceptually, 'create' and 'get' has their own purpose. 

'get' should fail if the requested env does not exist. 
It seems here 'create' uses caches automatically for anonymous envs and there is no need to delete anonymous envs. 

So, keep these two configs separate is fine with me.

**Review by jambayk — APPROVED** [review]
